pkg/controllers/cloud/account_poller.go

```diff
@@ -218,7 +224,38 @@ func (p *Poller) updateAccountPoller(name *types.NamespacedName, selector *cloud
 	return nil
 }
 
+// RestartAccountPoller stops and starts a goroutine making sure there is only one account poller goroutine at a time.
+func (p *Poller) RestartAccountPoller(name *types.NamespacedName) error {
+	p.mutex.Lock()
```

Comment:
what is this lock for?

Reply:
This is for accPollers Map which stores account poller per account.